fractal.py

```diff
@@ -124,15 +135,41 @@ def generate_fractal(model, c=None, size=pair_reader(int)(DEFAULT_SIZE),
   side = max(width, height)
   deltax = (side - width) / 2 # Centralize
   deltay = (side - height) / 2
-  img = pylab.array([
-    [func((2 * (col + deltax) / (side - 1) - 1) / zoom + cx,
-          (2 * (height - row + deltay) / (side - 1) - 1) / zoom + cy)
-      for col in range(width)]
-    for row in range(height)
-  ])
-
+  
+  # Start a timer
+  start = time.time()
+
+  # Create a pool of workers
+  numProcs = multiprocessing.cpu_count()
+  pool = multiprocessing.Pool(numProcs)
+  procs = []
+
+  # Fill our image with all zeros
+  img = pylab.zeros((height, width))
+
+  # Slice the image into rows
+  for row in range(height):
+    args = [(width, height, side, deltax, deltay, cx, cy, zoom, row, func, depth, c)]
+    p = pool.apply_async(generate_process, args)
+    procs.append(p)
+
+  # Collect results from worker procs
+  for row in range(height):
+    img[row, :] = procs[row].get()
+
+  # Print the timer
+  print 'Time taken:', time.time()-start
```

Comment:
There's no print statement in Python 3. There's no problem on keeping the "Time taken" print, but it need parentheses and the `print_function` also imported from `__future__` to work on both versions of Python. As logging doesn't seem like a problem, you can print the "Amount of procs" before the time taken, it should be useful as well.